include/deal.II/grid/grid_tools.h

```diff
@@ -1291,6 +1319,55 @@ namespace GridTools
 
 namespace GridTools
 {
+
+  // This anonymous namespace contains utility functions to extract the
+  // triangulation from any container such as DoFHandler and the like
+  namespace
+  {
+    template<int dim, int spacedim>
+    const Triangulation<dim, spacedim> &
```

Comment:
All of this could stay in the .cc file if you used the `std_cxx11::function` argument.